### PR TITLE
allow for non github repos

### PR DESCRIPTION
**Pull request by steveoh**
## Description of Changes

This pull requests allows for an object to be set in the repo's array

```json
"repositories": [{ 
	  "host": "https://gitlabs.com/",
	  "repo": "name/repo"
}]
```

### CODE REVIEW

File: src/forklift/engine.py [hunk 1]
```diff
@@ -567,19 +567,31 @@ def _send_report_email(template, report_object, subject):
 
 def _clone_or_pull_repo(repo_name):
     '''repo_name: string - a github repository username/reponame format
+                  or an object with host and repo with the username/reponame syntax
```
Comment (stdavis) [inline src/forklift/engine.py]: We should add examples of each of the accepted formats.

Comment (steveoh) [reply src/forklift/engine.py]: to here or the readme?

**Comment by steveoh** [comment]
I think the complicated part is going to be access tokens etc...

File: src/forklift/engine.py (hunk 1 above)
Comment (steveoh) [reply src/forklift/engine.py]: I added both...

**Comment by steveoh** [comment]
I think this is ready to merge

**Comment by stdavis** [comment]
I successfully tested this on a DTS GitLabs repo. Nice work!

**Pull request merged**